pkg/maps/nat/stats/cell.go

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package stats
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/cilium/cilium/pkg/maps/nat"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/metrics/metric"
+	"github.com/cilium/cilium/pkg/statedb"
+	"github.com/cilium/cilium/pkg/statedb/index"
+	"github.com/cilium/hive/cell"
+	"github.com/spf13/pflag"
+)
+
+var Cell = cell.Module(
+	"nat-stats",
+	"Aggregates stats for NAT maps",
+	metrics.Metric(newMetrics),
+	cell.ProvidePrivate(newTables),
+	cell.Provide(
+		func(m Metrics) natMetrics {
+			return m
+		},
+		newStats,
+	),
+	cell.Config(Config{
+		NATMapStatInterval:       30 * time.Second,
+		NatMapStatKStoredEntries: 32,
+	}),
+	cell.Invoke(func(_ *Stats) {}),
+)
+
+type Metrics struct {
+	LocalPorts metric.Vec[metric.Gauge]
+}
+
+type natMetrics interface {
+	updateLocalPorts(family nat.IPFamily, count, maxPorts int)
+}
+
+func (m Metrics) updateLocalPorts(family nat.IPFamily, count, maxPorts int) {
+	m.LocalPorts.WithLabelValues(family.String()).Set(float64(count) / float64(maxPorts))
+}
+
+func newMetrics() Metrics {
+	return Metrics{
+		LocalPorts: metric.NewGaugeVecWithLabels(metric.GaugeOpts{
```

Comment:
Can we document `nat_endpoint_max_connection` here?
https://github.com/cilium/cilium/blob/579d3b624444792c8b65002c5c7f284f9d98ec90/Documentation/observability/metrics.rst